HHS/Head-Start-TTADP
# [TTAHUB-2888] Fix saving duplicate aro topics

## Description of change

This seems to be an edge case for a specific report with multiple recipients. This change prevents saving duplicate ARO topics for the same objective by moving the race condition functions outside of the loop.

## How to test

- Create a report test saving/updating the objective topics / files / and resources (make sure changes are correctly reflected in the db).
- On the prod db add a recipient to report id: 39038 it should save without error.


## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-0


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
